digitalpy/core/zmanager/integration_manager.py

```diff
@@ -11,33 +11,50 @@
 import zmq
 
 class IntegrationManager:
+    def __init__(self, integration_manager_puller_protocol: str, integration_manager_puller_address: str, integration_manager_puller_port: int,
+                 integration_manager_publisher_protocol: str, integration_manager_publisher_address: str, integration_manager_publisher_port: int) -> None:
+        self.integration_manager_puller_protocol = integration_manager_puller_protocol
+        self.integration_manager_puller_address = integration_manager_puller_address
+        self.integration_manager_puller_port = integration_manager_puller_port
+        self.integration_manager_publisher_protocol = integration_manager_publisher_protocol
+        self.integration_manager_publisher_address = integration_manager_publisher_address
+        self.integration_manager_publisher_port = integration_manager_publisher_port
+
+
     def initialize_connections(self):
         context = zmq.Context()
 
         # create a pull socket
         pull_socket = context.socket(zmq.PULL)
-        pull_socket.setsockopt(zmq.RCVHWM, 1000)  # limit the number of unprocessed requests that can be queued
-        pull_socket.bind("tcp://127.0.0.1:12345")
+        # unlimited as trunkating can result in unsent data and broken messages
+        # TODO: determine a sane default
+        pull_socket.setsockopt(zmq.RCVHWM, 0)
+        pull_socket.bind(f"{self.integration_manager_puller_protocol}://{self.integration_manager_puller_address}:{self.integration_manager_puller_port}")
         self.pull_socket = pull_socket
 
         # create a pub socket
         pub_socket = context.socket(zmq.PUB)
-        pub_socket.setsockopt(zmq.SNDHWM, 1000)  # limit the number of unsent responses that can be queued
-        pub_socket.bind("tcp://127.0.0.1:12346")
+        # unlimited as trunkating can result in unsent data and broken messages
+        # TODO: determine a sane default
+        pub_socket.setsockopt(zmq.SNDHWM, 0)
+        pub_socket.bind(f"{self.integration_manager_publisher_protocol}://{self.integration_manager_publisher_address}:{self.integration_manager_publisher_port}")
         self.pub_socket = pub_socket
 
-    def main(self):
+    def start(self):
+        """this is the main running function for the integration manager"""
```

Comment:
**[pep8]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>D401 First line should be in imperative mood; try rephrasing